server/src/main/java/org/elasticsearch/action/bulk/BulkRequestParser.java

```diff
@@ -184,6 +185,7 @@ public void parse(
                 }
                 String action = parser.currentName();
                 if (SUPPORTED_ACTIONS.contains(action) == false) {
+                    assert Version.CURRENT.major == Version.V_7_17_0.major + 1;
```

Comment:
Why not just `assert Version.CURRENT.major == Version.V_8_0_0.major;` ?

Reply:
The point is to mention a symbol that will be removed in v9 so that this is a compile error (the assertion failure is kinda not the point)